libgoal/teal.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2019-2020 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package libgoal
+
+import (
+	"github.com/algorand/go-algorand/crypto"
+)
+
+// Compile compiles the given program and returned the compiled program
+func (c *Client) Compile(program []byte) (compiledProgram []byte, compiledProgramHash crypto.Digest, err error) {
+	algod, err2 := c.ensureAlgodClient()
+	if err2 != nil {
+		return nil, crypto.Digest{}, err2
+	}
+	var hash []byte
+	compiledProgram, hash, err = algod.Compile(program)
```

Comment:
As I understand `compiledProgram` is base64 and `hash` is string hash, so conversion needed here or in the test

Reply:
see the handler:
```
	response := generated.PostCompileResponse{
		Hash:   addr.String(),
		Result: base64.StdEncoding.EncodeToString(program),
	}
```

Reply:
I've added the conversion in the restClient.go, since this is where it leaves the "http tier". Both base64 and string formatting are meant for the way to tunnel these over http, so that's the best place to decode these as well.